### ISSUE TITLE
Failed to watch *v2beta1.MPIJob:P failed to list *v2beta1.MPIJob: ...

### ISSUE THREAD

**Issue by thuzhf** [kubeflow/mpi-operator#473]
Sometimes I encounter the following log in `deploy/mpi-operator`, and the operator does not respond to my apply request.
![image](https://user-images.githubusercontent.com/8064242/186067243-2d93c1ec-2397-4fd4-be4a-fba308c5a7df.png)

Is this a bug or what?

**Comment by gaocegege** [comment]
cc @zw0610

**Comment by zw0610** [comment]
Could you post the yaml file of `MPIJob` that posted? Seems like some issue with the `EnvVar` field in the `MPIJob` that the value of an environment variable cannot be set as  a boolean value `true` but a string value `"true"`?

**Comment by thuzhf** [comment]
@zw0610 My yaml is as follows:
![image](https://user-images.githubusercontent.com/8064242/186070645-d71c983e-1f1a-40e4-b7c8-c8dd9bca4e36.png)

**Comment by thuzhf** [comment]
> Sometimes I encounter the following log in `deploy/mpi-operator`, and the operator does not respond to my apply request. ![image](https://user-images.githubusercontent.com/8064242/186067243-2d93c1ec-2397-4fd4-be4a-fba308c5a7df.png)
> 
> Is this a bug or what?

And when this happens, even if I rollout restart `deploy/mpi-operator`,  it still does not recover from this condition.

**Comment by zw0610** [comment]
> @zw0610 My yaml is as follows: ![image](https://user-images.githubusercontent.com/8064242/186070645-d71c983e-1f1a-40e4-b7c8-c8dd9bca4e36.png)

could you post all `EnvVar` parts in the `MPIJob` rather than the cropped one?

**Comment by thuzhf** [comment]
> > @zw0610 My yaml is as follows: ![image](https://user-images.githubusercontent.com/8064242/186070645-d71c983e-1f1a-40e4-b7c8-c8dd9bca4e36.png)
> 
> could you post all `EnvVar` parts in the `MPIJob` rather than the cropped one?

This is already the full yaml content. I do not write any EnvVar by myself in this yaml file.

**Comment by zw0610** [comment]
Could you get from APIServer via `kubectl` instead of the source file you posted? There might be some legacy jobs or mutating webhook that makes the job in the cluster different from the source file.

**Comment by thuzhf** [comment]
This is the result of `kubectl get pod simple-mpi-worker-1 -o yaml`:
![image](https://user-images.githubusercontent.com/8064242/186074157-2442d2b7-2bc6-4a8e-8d15-fcd2f8448052.png)

**Comment by thuzhf** [comment]
And this is the result of `kubectl get pod simple-mpi-launcher-xxxx`:
![image](https://user-images.githubusercontent.com/8064242/186074933-f84d3ee0-a90d-47e4-9d86-650dd67d9be0.png)

**Comment by zw0610** [comment]
I'm afraid I cannot link this mpijob to the error message on the top unless there are other mpijobs.

**Comment by alculquicondor** [comment]
Is this the only MPIJob in the cluster?

You wouldn't have pods for this MPIJob, because it fails to parse.

**Comment by thuzhf** [comment]
> Is this the only MPIJob in the cluster?

No. But even if someone else's MPIJob caused this. Why does this cause the whole `mpi-operator` to not work? I think it is not expected to not work because someone else apply a wrong yaml. Do I miss something?

**Comment by alculquicondor** [comment]
It's how the k8s go client works: it needs to fetch all MPIJobs in the cluster. If one of them doesn't parse, it fails.

I'm not sure if there is a way to skip the failing one. A quick google search didn't give me positive results.

**Comment by alculquicondor** [comment]
In general, we would avoid these problems by having a validation webhook. Unfortunately, we haven't set up one yet.

**Comment by thuzhf** [comment]
OK. But how can I know the full yaml content of the corresponding yaml to trace it? From the log, I have no idea what yaml/whose yaml caused this log. If I can know the full content of the corresponding yaml, I can easily find the creator of this yaml.

**Comment by alculquicondor** [comment]
I would suggest using `kubectl get -A mpijobs` and grep for `OMPI_ALLOW_RUN_AS_ROOT`

**Comment by thuzhf** [comment]
Understood. Thanks!
